### PR TITLE
[17.0][MIG] remove_odoo_enterprise: Migration to 17.0

### PR DESCRIPTION
**Pull request by remi-filament**

### CODE REVIEW

**Comment by vincent-hatakeyama** [comment]
Some features are not working correctly:

- in apps, all the odoo enterprise modules are still displayed.

The feature that work are:

- removal of the enterprise features from settings
- the app stores button is removed

File: remove_odoo_enterprise/models/ir_module_module.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from odoo import api, models
+
+
+class IrModuleModule(models.Model):
+    _inherit = "ir.module.module"
+
+    @api.model
+    def search(self, domain, offset=0, limit=None, order=None):
+        domain += [("to_buy", "=", False)]
+        return super().search(domain, offset, limit, order)
```
Comment (vincent-hatakeyama) [inline remove_odoo_enterprise/models/ir_module_module.py]:
```suggestion
    def search_fetch(self, domain, field_names, offset=0, limit=None, order=None):
        domain += [("to_buy", "=", False)]
        return super().search_fetch(domain, field_names, offset, limit, order)
```

With these changes, the enterprise modules are filtered.

**Review by vincent-hatakeyama — CHANGES_REQUESTED** [review]

**Comment by remi-filament** [comment]
You are right @vincent-hatakeyama , thank you for checking that !
Sorry, I did not spotted it locally, since I realize to_buy modules where removed from my local database...

Web client does not use search() function anymore but web_search_read (since https://github.com/odoo/odoo/pull/114024 and web_search_read was first added as unity_web_search_read with https://github.com/odoo/odoo/pull/119034)

Here, I asked myself whether we should override web_search_read or search_fetch as you propose, and I chose the second one so that it has broader impact than just on the web interface. Also, I merged the check done for payment provider and ir_module_module in the same base.py file since having it in 2 different places causes 2 overrides when searching in ir_module_module.

**Review by DorianMAG — APPROVED** [review]
Tested in runboat.
LGTM
Thx for this work

**Comment by carolinafernandez-tecnativa** [comment]
Functional review LGTM, could you please squash administrative commits? 
Check https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate for further information

**Comment by carolinafernandez-tecnativa** [comment]
Hi! Can you check my comment before ? so we can move forward with this migration :) Thanks!

**Comment by remi-filament** [comment]
Thanks for your review @carolinafernandez-tecnativa I have squashed bot commits.

**Review by carolinafernandez-tecnativa — APPROVED** [review]
LGTM Thanks :)

**Comment by carolinafernandez-tecnativa** [comment]
@pedrobaeza do you think that can we proceed to merge?

**Comment by carolinafernandez-tecnativa** [comment]
@remi-filament just a little comment, please rename PR with OCA conventions. It should be named `[17.0][MIG] <module>: Migration to 17.0`. 

Check this https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0 for further information.

**Pull request merged**